src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -233,6 +233,13 @@ protected void AddSecrets(TransactionBuildContext context)
         /// <param name="context">The context associated with the current transaction being built.</param>
         protected void FindChangeAddress(TransactionBuildContext context)
         {
+            if (context.ChangeAddress != null)
```

Comment:
Just a minor nit, you could change it around a little to avoid the early return:

```
if (context.ChangeAddress == null)
{
    // Get an address to send the change to.
    context.ChangeAddress = this.walletManager.GetUnusedChangeAddress(new WalletAccountReference(context.AccountReference.WalletName, context.AccountReference.AccountName));
}

context.TransactionBuilder.SetChange(context.ChangeAddress.ScriptPubKey);
```